File: benchmarks/fib/fib.rs
fn fib(n: i32) -> i32 {
    let mut a = 0i32;
    let mut b = 1i32;
    for _ in 0..n {
        let tmp = a + b;
        a = b;
        b = tmp;
    }
    b
}

fn main() {
    println!("Result: {}", fib(500_000_000));
}

File: benchmarks/fib/fib.go
package main

func fib(n int32) int32 {
	var a, b int32 = 0, 1
	for i := int32(0); i < n; i++ {
		tmp := a + b
		a = b
		b = tmp
	}
	return b
}

func main() {
	println("Result:", fib(500000000))
}
